src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -1134,3 +1165,68 @@ func isRedpandaClusterVersionManaged(
 	}
 	return true
 }
+
+func updateStatus(ctx context.Context, client client.Client, observedCluster *vectorizedv1alpha1.Cluster, mutator func(cluster *vectorizedv1alpha1.Cluster)) (vectorizedv1alpha1.ClusterStatus, error) {
+	var result vectorizedv1alpha1.ClusterStatus
+	name := types.NamespacedName{
+		Namespace: observedCluster.Namespace,
+		Name:      observedCluster.Name,
+	}
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I think it would be better to use client.Patch rather than retry mechanism, but in our case it shouldn't matter.